### DISCUSSION TITLE
React compiler not working

### DISCUSSION THREAD

**Discussion by yanglee2421** [vitejs/vite-plugin-react#361] [Q&A] [answered]
The includes contain .scss, which causes the react compiler (a babel plugin) to not work. Why?
```tsx
export default defineConfig(() => {
  return {
    plugins: [
      react({
        babel: {
          plugins: [["babel-plugin-react-compiler", {}]],
        },
        include: [".scss"],
      }),
    ],
  };
});
```

**Comment by ArnaudBarre** [comment]
Why do you want non JS files to be processed by Babel?

**Reply by yanglee2421** [reply]
When using .module.scss, if you do not include .scss in the includes option, fast refresh will cause the old styles to be lost and the new styles will not work when the scss file is modified.

**Reply by ArnaudBarre** [reply]
Can you provide a repro? CSS modules HMR works well and SCSS is just pre-transform from Vite point of view.

**Reply by yanglee2421** [reply]
https://github.com/yanglee2421/react-compiler-css-module
## Reproduction method
Change red in styles.module.css to green, or change 400 in styles.module.scss to 700.

This seems to be caused by the automatic memoization of the react compiler.

**Reply by hi-ogawa** [reply]
Thanks for the reproduction. It looks like the issue is not only about css module but it happens for any non-self-accepting hmr change. Simpler version would be https://stackblitz.com/github/hi-ogawa/react-compiler-css-module/tree/chore-minimal-repro?file=src%2Frepro.tsx

```tsx
// repro.tsx
export default "hello"; // changing this doesn't update App.tsx

// App.tsx
import repro from "./repro";

export default function App() {
  return (
    <p>repro: {repro}</p>
  );
}
```

I'm not sure if there's something Vite plugin needs to handle for hmr or it's simply a bug on react-compiler side. While fiddling with it, I also found that removing `<StrictMode>` fixes the issue.

**Reply by ArnaudBarre** [reply]
Oh WTF this is so bad. I need to investigate

**Reply by ArnaudBarre** [reply, marked as answer]
React issue: https://github.com/facebook/react/issues/29115
